File: imu_calib_matlab/tests/test_allan.m
% test_allan
% Validate Allan analysis modules on the synthetic static dataset.

clear;
clc;

projectRoot = fileparts(fileparts(mfilename('fullpath')));
addpath(genpath(fullfile(projectRoot, 'src')));

[data, ~] = load_example_data('forceRegenerate', true, 'saveToMat', false);

gyroAllan = analyze_gyro_allan(data.static.t, data.static.gyro);
accAllan = analyze_acc_allan(data.static.t, data.static.acc);

assert(isfield(gyroAllan, 'valid') && isfield(gyroAllan, 'message'), ...
    'Gyro Allan output missing required status fields.');
assert(isfield(accAllan, 'valid') && isfield(accAllan, 'message'), ...
    'Accel Allan output missing required status fields.');

if gyroAllan.valid
    assert(~isempty(gyroAllan.tau) && ~isempty(gyroAllan.adev), ...
        'Valid gyro Allan result must contain tau and adev.');
end

if accAllan.valid
    assert(~isempty(accAllan.tau) && ~isempty(accAllan.adev), ...
        'Valid accel Allan result must contain tau and adev.');
end

fprintf('test_allan passed.\n');

File: imu_calib_python/tests/test_allan.py
from imu_calib.calib.analyze_acc_allan import analyze_acc_allan
from imu_calib.calib.analyze_gyro_allan import analyze_gyro_allan
from imu_calib.io.load_example_data import load_example_data


def test_allan_outputs_are_valid_for_example_static_data():
    data, _, _ = load_example_data()
    gyro_allan = analyze_gyro_allan(data.static.t, data.static.gyro)
    acc_allan = analyze_acc_allan(data.static.t, data.static.acc)

    assert gyro_allan["valid"] is True
    assert acc_allan["valid"] is True
    assert gyro_allan["tau"].ndim == 1
    assert gyro_allan["adev"].shape[1] == 3
    assert "estimate" in gyro_allan
